tests/setup_teardown.rs

```diff
@@ -0,0 +1,113 @@
+use httpmock::Method::{GET, POST};
+use httpmock::{mock, with_mock_server};
+
+mod common;
+
+use goose::prelude::*;
+
+const INDEX_PATH: &str = "/";
+const SETUP_PATH: &str = "/setup";
+const TEARDOWN_PATH: &str = "/teardown";
+
+pub async fn setup(client: &GooseClient) {
+    let _response = client.post(SETUP_PATH, "setting up load test").await;
+}
+
+pub async fn teardown(client: &GooseClient) {
+    let _response = client
+        .post(TEARDOWN_PATH, "cleaning up after load test")
+        .await;
+}
+
+pub async fn get_index(client: &GooseClient) {
+    let _response = client.get(INDEX_PATH).await;
+}
+
+/// Test test_start alone.
+#[test]
+#[with_mock_server]
+fn test_start() {
+    let mock_setup = mock(POST, SETUP_PATH).return_status(201).create();
+    let mock_teardown = mock(POST, TEARDOWN_PATH).return_status(205).create();
+    let mock_index = mock(GET, INDEX_PATH).return_status(200).create();
+
+    crate::GooseAttack::initialize_with_config(common::build_configuration())
+        .setup()
+        .test_start(task!(setup))
+        .register_taskset(taskset!("LoadTest").register_task(task!(get_index).set_weight(9)))
+        .execute();
+
+    let called_setup = mock_setup.times_called();
+    let called_index = mock_index.times_called();
+    let called_teardown = mock_teardown.times_called();
+
+    // Confirm the load test ran.
+    assert_ne!(called_index, 0);
```

Comment:
Comment does not match the assertion.

Reply:
If the test doesn’t run, this will be zero. If the test runs it will not be zero. We are confirming the test runs by confirming the mock endpoint served more than 0 requests. 

Comment does match assertion.